lib/src/Base/Optim/NLopt.cxx

```diff
@@ -154,8 +154,6 @@ void NLopt::checkProblem(const OptimizationProblem & problem) const
 {
   if (problem.hasMultipleObjective())
     throw InvalidArgumentException(HERE) << "Error: " << getAlgorithmName() << " does not support multi-objective optimization";
-  if (problem.hasResidualFunction())
```

Comment:
Why can this check be removed?

Reply:
we can let nlopt solve problem eventhough it only sees the total square residual and not the residual per-output